### PR TITLE
[BI-1074] Add multiple error display and breeding method errors for germplasm import

### PR DESCRIPTION
**Pull request by davedrp**
# Description
**Story:** [BI-1074 - 1.4 Germplasm Import - Validate Breeding Methods](https://breedinginsight.atlassian.net/browse/BI-1074)

When Importing Germplasms, validate the breading methods against the breeding_method table.

# Dependencies
Must be run with bi-api branch _feature/BI-1074_.

# Checklist:

- [ x] I have performed a self-review of my own code
- [ x] I have tested my code and ensured it meets the acceptance criteria of the story
- [ ] I have create/modified unit tests to cover this change
- [ x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to documentation
- [ ] I have run TAF.

### CODE REVIEW

File: src/breeding-insight/service/ImportService.ts [hunk 1]
```diff
@@ -120,4 +122,22 @@ export class ImportService {
...
+      if (rowErrors) {
+        let validationError: ValidationError = new ValidationError(rowErrors);
+        return validationError;
+      }
+    } else {
+      return "Cannot find error";
+    }
+    // Should not get here
+    return null;
+  }
```
Comment (ctucker3) [inline src/breeding-insight/service/ImportService.ts:142]: Right now you are returning:

```
progress: {
  errors: {
    rowErrors: [
      <errors>
    ]
  }
}
```

Does `errors` have a purpose? Could you remove `errors` and just return this like the traits import: 

```
progress: {
    rowErrors: [
      <errors>
    ]
}
```

**Review by ctucker3 — CHANGES_REQUESTED** [review]
The row numbers seem to be wrong. My errors are on row 2+ in this file, but the row numbers in the message start at 1. Here is my file: 
[BI-1074_bad_methods.xls](https://github.com/Breeding-Insight/bi-web/files/7891474/BI-1074_bad_methods.xls)

Here is the error: 
screencapture-localhost-8080-programs-af970a5a-2b34-4225-b5b8-7dff83c18190-import-germplasm-2022-01-18-13_16_14

**Review by nickpalladino — APPROVED** [review]

**Comment by ctucker3** [comment]
Also need to run TAF, so I'll approve after the title and TAF.

**Comment by davedrp** [comment]
> Also need to run TAF, so I'll approve after the title and TAF.

Done

**Review by ctucker3 — APPROVED** [review]

**Pull request merged**